mlrun/serving/routers.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import mlrun
+
+from io import BytesIO
+
+
+class BaseModelRouter:
+    """base model router class"""
+
+    def __init__(self, context, state, **kwargs):
+        self.context = context
+        self.routes = {}
+        for key, child in state.items():
+            if hasattr(child, "run"):
+                child = child.run
+            self.routes[key] = child
+        self.url_prefix = kwargs.get("url_prefix", "/v2/models")
+        self.health_prefix = kwargs.get("health_prefix", "/v2/health")
+        self.protocol = kwargs.get("protocol", "v2")
+        self.inputs_key == "instances" if self.protocol == "v1" else "inputs"
+
+    def parse_event(self, event):
+        parsed_event = {}
+        try:
+            if not isinstance(event.body, dict):
+                body = json.loads(event.body)
+            else:
+                body = event.body
+            if "data_url" in body:
+                # Get data from URL
+                url = body["data_url"]
+                self.context.logger.debug_with("downloading data", url=url)
```

Comment:
this is Nuclio context, but anyway i will change to make it genetic

Reply:
I see, if it's Nuclio context then it's Nuclio logger, and maybe you do need the `_with` cause you're adding variables, pleace check that

Reply:
checked, change back to `debug_with`

Reply:
@Hedingber the logger can be either Nuclio or mlrun (in case of local run), so i rather just keep it the way it is, if u want i can look up an article from Uber (who implemented this .debug_with) which explains why its all BS

Reply:
If you want to support both you shouldn't use `url=url`, you should use string formatting

Reply:
@Hedingber done